langs/clojure.go

```diff
@@ -0,0 +1,249 @@
+package langs
+
+import (
+    "bytes"
+    "encoding/json"
+    "errors"
+    "fmt"
+    "io/ioutil"
+    "net/http"
+    "os"
+    "path/filepath"
+)
+
+// ClojureLangHelper provides a set of helper methods for the lifecycle of Clojure Leiningen projects
+type ClojureLangHelper struct {
+    BaseHelper
+    latestFdkVersion string
+    version          string
+}
+
+func (h *ClojureLangHelper) Handles(lang string) bool {
+    for _, s := range h.LangStrings() {
+        if lang == s {
+            return true
+        }
+    }
+    return false
+}
+func (h *ClojureLangHelper) Runtime() string {
+    return h.LangStrings()[0]
+}
+
+func (lh *ClojureLangHelper) LangStrings() []string {
+    return []string{"clojure"}
+
+}
+func (lh *ClojureLangHelper) Extensions() []string {
+    return []string{".clj"}
+}
+
+func (lh *ClojureLangHelper) BuildFromImage() (string, error) {
+    return "clojure:lein", nil
+}
+
+// RunFromImage returns the Docker image used to run the Clojure function.
+func (lh *ClojureLangHelper) RunFromImage() (string, error) {
+    return "openjdk:9-jre", nil
+}
+
+// HasBoilerplate returns whether the Java runtime has boilerplate that can be generated.
+func (lh *ClojureLangHelper) HasBoilerplate() bool { return true }
+
+// Clojure defaults to CloudEvents
+func (lh *ClojureLangHelper) DefaultFormat() string { return "cloudevent" }
+
+// GenerateBoilerplate will generate function boilerplate for a Java runtime.
+// The default boilerplate is for a Leiningen project.
+func (lh *ClojureLangHelper) GenerateBoilerplate(path string) error {
+    pathToProjectFile := filepath.Join(path, "project.clj")
+    if exists(pathToProjectFile) {
+        return ErrBoilerplateExists
+    }
+
+    apiVersion, err := lh.getFDKAPIVersion()
+    if err != nil {
+        return err
+    }
+
+    if err := ioutil.WriteFile(pathToProjectFile,
+        []byte(clojureProjFileContent(lh.version, apiVersion)),
+        os.FileMode(0644)); err != nil {
+        return err
+    }
+
+    pathToTestJson := filepath.Join(path, "test.json")
+    if exists(pathToTestJson) {
+        return ErrBoilerplateExists
+    }
+    if err := ioutil.WriteFile(pathToTestJson,
+        []byte(fnTestBoilerplate),
+        os.FileMode(0644)); err != nil {
+        return err
+    }
+
+    mkDirAndWriteFile := func(dir, filename, content string) error {
+        fullPath := filepath.Join(path, dir)
+        if err = os.MkdirAll(fullPath, os.FileMode(0755)); err != nil {
+            return err
+        }
+
+        fullFilePath := filepath.Join(fullPath, filename)
+        return ioutil.WriteFile(fullFilePath, []byte(content), os.FileMode(0644))
+    }
+
+    err = mkDirAndWriteFile("src/example", "core.clj", helloClojureSrcBoilerplate)
```

Comment:
let's call it `src/func`, we `func` across multiple lang helpers, so, let's stick with that